master/buildbot/reporters/http.py

```diff
@@ -45,58 +72,27 @@ def reconfigService(self, builders=None, debug=None, verify=None,
         self.wantPreviousBuild = wantPreviousBuild
         self.wantLogs = wantLogs
 
-    @defer.inlineCallbacks
-    def startService(self):
-        yield super().startService()
+        if generators is None:
+            generators = self._create_generators_from_old_args(builders, wantProperties, wantSteps,
+                                                               wantPreviousBuild, wantLogs)
 
-        startConsuming = self.master.mq.startConsuming
-        self._buildCompleteConsumer = yield startConsuming(
-            self.buildFinished,
-            ('builds', None, 'finished'))
+        yield super().reconfigService(generators=generators)
 
-        self._buildStartedConsumer = yield startConsuming(
-            self.buildStarted,
-            ('builds', None, 'new'))
+    def _create_generators_from_old_args(self, builders, want_properties, want_steps,
+                                         want_previous_build, want_logs):
+        formatter = MessageFormatterEmpty(wantProperties=want_properties, wantSteps=want_steps,
+                                          wantLogs=want_logs)
+        return [
+            BuildStatusGenerator(builders=builders, message_formatter=formatter, report_new=True,
+                                 _want_previous_build=want_previous_build)
+        ]
 
-    def stopService(self):
-        self._buildCompleteConsumer.stopConsuming()
-        self._buildStartedConsumer.stopConsuming()
+    def sendMessage(self, reports):
+        return self.send(reports[0]['builds'][0])
 
-    @defer.inlineCallbacks
-    def _got_event(self, key, msg):
-        # This function is used only from tests
-        if key[0] != 'builds':  # pragma: no cover
-            raise Exception('Invalid key for _got_event: {}'.format(key))
-        if key[2] == 'new':
-            yield self.buildStarted(key, msg)
-        elif key[2] == 'finished':
-            yield self.buildFinished(key, msg)
-        else:  # pragma: no cover
-            raise Exception('Invalid key for _got_event: {}'.format(key))
-
-    def buildStarted(self, key, build):
-        return self.getMoreInfoAndSend(build)
-
-    def buildFinished(self, key, build):
-        return self.getMoreInfoAndSend(build)
-
-    def filterBuilds(self, build):
-        if self.builders is not None:
-            return build['builder']['name'] in self.builders
-        return True
-
-    @defer.inlineCallbacks
-    def getMoreInfoAndSend(self, build):
-        yield utils.getDetailsForBuild(self.master, build, wantProperties=self.wantProperties,
-                                       wantSteps=self.wantSteps,
-                                       wantPreviousBuild=self.wantPreviousBuild,
-                                       wantLogs=self.wantLogs)
-        if self.filterBuilds(build):
-            yield self.send(build)
-
-    @abc.abstractmethod
-    def send(self, build):
-        pass
+    # Deprecated overridden method, will be removed in Buildbot 3.0
```

Comment:
looks like an overlook

Reply:
I've added this as a documentation that subclasses may implement this method and we need to handle it.